grails-app/controllers/org/pih/warehouse/picklist/PicklistController.groovy

```diff
@@ -94,28 +94,54 @@ class PicklistController {
         render(template: "/picklist/print", model: [requisition: requisition, picklist: picklist, location: location, order: params.order])
     }
 
+    private Map formatPicklistItemPickExport(Map item) {
+        return [
+                "${g.message(code: 'default.id.label')}": item?.id ?: "",
+                "${g.message(code: 'product.productCode.label')}": item?.productCode ?: "",
+                "${g.message(code: 'product.name.label')}": item?.productName ?: "",
+                "${g.message(code: 'inventoryItem.lotNumber.label')}": item?.lotNumber ?: "",
+                "${g.message(code: 'inventoryItem.expirationDate.label')}": item?.expirationDate
+                        ? item.expirationDate?.format(Constants.EXPIRATION_DATE_FORMAT)
+                        : "",
+                "${g.message(code: 'inventoryItem.binLocation.label')}": item?.binLocation ?: "",
+                "${g.message(code: 'picklist.quantity.label')}": item?.quantity == null ? "" : item?.quantity,
+        ]
+    }
+
     def exportPicklistItems() {
         String format = params.get("format", "csv")
 
         List<PickPageItem> pickPageItems = stockMovementService.getPickPageItems(params.id, null, null)
-        List<PicklistItem> picklistItems = pickPageItems.picklistItems?.flatten()
 
         // We need to create at least one row to ensure an empty template
-        if (picklistItems?.empty) {
-            picklistItems.add(new PicklistItem())
+        if (pickPageItems?.empty) {
+            pickPageItems.add(new PickPageItem())
         }
 
-        def lineItems = picklistItems.collect {
-            [
-                "${g.message(code: 'default.id.label')}": it?.requisitionItem?.id ?: "",
-                "${g.message(code: 'product.productCode.label')}": it?.requisitionItem?.product?.productCode ?: "",
-                "${g.message(code: 'product.name.label')}": it?.requisitionItem?.product?.name ?: "",
-                "${g.message(code: 'inventoryItem.lotNumber.label')}": it?.inventoryItem?.lotNumber ?: "",
-                "${g.message(code: 'inventoryItem.expirationDate.label')}": it?.inventoryItem?.expirationDate ? it.inventoryItem.expirationDate.format(Constants.EXPIRATION_DATE_FORMAT) : "",
-                "${g.message(code: 'inventoryItem.binLocation.label')}": it?.binLocation?.name ?: "",
-                "${g.message(code: 'default.quantity.label')}": it?.quantity ?: "",
+
+        List<Map> lineItems = pickPageItems.collectMany { pickPageItem ->
+            if (pickPageItem.picklistItems.size() > 0) {
+                return pickPageItem.picklistItems.collect { picklistItem ->
+                    return [
+                            id: picklistItem?.requisitionItem?.id,
+                            productCode: picklistItem?.requisitionItem?.product?.productCode,
+                            productName: picklistItem?.requisitionItem?.product?.name,
```

Comment:
I'm wondering what about synonyms here, if we want to include them or not, so to return `displayName ?: name`?